File: Javascript/Two-Sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
const twoSum = (nums, target) => {
  let targetMap = new Map();

  for (let i = 0; i < nums.length; i++) {
    if (targetMap.has(target - nums[i])) {
      return [i, targetMap.get(target - nums[i])];
    }

    targetMap.set(nums[i], i);
  }
};

File: Dart/Two-Sum.dart
class Solution {
  List<int> twoSum(List<int> nums, int target) {
    var numsMap = Map<int, int>();

    for (int i = 0; i < nums.length; i++) {
      var diff = target - nums[i];
      if (numsMap.containsKey(diff)) return [numsMap[diff]!, i];
      numsMap[nums[i]] = i;
    }

    throw "";
  }
}
